test/e2e/presto_prometheus_connector_up_test.go

```diff
@@ -0,0 +1,50 @@
+package e2e
+
+import (
+	"bytes"
+	"github.com/operator-framework/operator-metering/test/reportingframework"
+	"github.com/stretchr/testify/require"
+	"os"
+	"os/exec"
+	"strings"
+
+	"testing"
+)
+
+func testPrometheusConnectorWorks(t *testing.T, rf *reportingframework.ReportingFramework) {
+	t.Helper()
+	out, err := exec.Command(
+		"/usr/local/bin/kubectl",
+		"-n", rf.Namespace,
+		"get", "pods",
+		"-l", "app=presto",
+		"-o", "name",
+	).Output()
+	require.Nil(t, err, "expected querying for the presto pod name would produce no error")
+	require.NotEmpty(t, string(out), "unable to parse presto pod name")
+	// we know the output is going to be of the form `pod/<presto coordinator pod name>` so split by '/'
+	tmp := strings.Split(string(out), "/")[1]
+	host := strings.TrimSuffix(tmp, "\n")
+	cmd := exec.Command(
+		"/usr/local/bin/kubectl",
+		"-n", rf.Namespace,
+		"exec",
+		host,
+		"-c", "presto",
+		"--",
+		"presto-cli",
+		"--server", "https://presto:8080",
+		"--user", "root",
+		"--catalog", "prometheus",
+		"--schema", "default",
+		"--keystore-path", "/opt/presto/tls/keystore.pem",
+		"--execute",
+		"SELECT * FROM prometheus.default.up where timestamp > (NOW() - INTERVAL '10' second)",
+	)
+	var queryResults bytes.Buffer
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = &queryResults
+	err = cmd.Run()
+	queryResultsStr := string(queryResults.Bytes())
+	require.Containsf(t, queryResultsStr, "endpoint=metrics", "query output didn't contain `endpoint=metrics`")
```

Comment:
Let's add a check for a nil error too, e.g.:
```go
require.Nil(t, err, "expected that querying the prometheus catalog would produce no error)
```

Reply:
I thought about that originally, but you can't get `Containsf` without that also being true.

Reply:
Oh maybe, could you get some kind of err but still get results on the query? Perhaps, but it might be a warning or something. If you get the results and they Containsf that string, I think testing the other could unnecessarily fail the test.

Reply:
I feel like disregarding an error, despite the query returning results, may open us up to bugs in the future. I'm not sure though, as your point about `Containsf` is valid.

Reply:
I think to an extent, looser is better on an e2e test. To my mind e2e tests tell you if the system "generally" works, could still have some bugs, doesn't try to rule them all out. Especially on an e2e with a heavy setup/teardown time like all of these, I think it's better to get them to err of the side of working most of the time.

I don't feel the same about unit or integration tests, fwiw.